File: Regression/NRG/Landing/Landing_Links_Logo.rb
require 'rubygems'
require 'watir-webdriver'
require "win32ole"

### FOOTER ###
excel= WIN32OLE::new("excel.Application")
workbook=excel.Workbooks.Open("C:\\Scripts\\NRG\\Landing\\nrg_links.xlsx")
worksheet = workbook.worksheets(1)

browser = Watir::Browser.new :firefox

rows = 2		#2
while    
rows <= 5		#5

browser.goto 'http://enroll.pt.nrghomepower.com/combined/cashbackres/il/'

page=worksheet.cells(rows,"A").value
xpath=worksheet.cells(rows,"B").value
expected=worksheet.cells(rows,"C").value
result=worksheet.cells(rows,"E").value

string_from_excel = "goto"
browser.send string_from_excel, page
browser.link(:xpath => xpath).click

if
browser.alert.exists?
browser.alert.ok
end

if
browser.text.include? 'Server Error'
print "Server Error, "
end

if
browser.url == expected
print result, " = good", "\n"
else
print result, " = bad", "\n"
end
rows=rows+1
end

$end
workbook.Save
workbook.Close

$end
sleep 2

### LOGO ###
excel= WIN32OLE::new("excel.Application")
workbook=excel.Workbooks.Open("C:\\Scripts\\NRG\\Landing\\nrg_links.xlsx")
worksheet = workbook.worksheets(1)

rows = 6		#6
while    
rows <= 6		#6

browser.goto 'http://enroll.pt.nrghomepower.com/combined/cashbackres/il/'

page=worksheet.cells(rows,"A").value
xpath=worksheet.cells(rows,"B").value
expected=worksheet.cells(rows,"C").value
result=worksheet.cells(rows,"E").value

string_from_excel = "goto"
browser.send string_from_excel, page
browser.image(:xpath => xpath).click

if
browser.alert.exists?
browser.alert.ok
end

if
browser.text.include? 'Server Error'
print "Server Error, "
end

if
browser.url == expected
print result, " = good", "\n"
else
print result, " = bad", "\n"
end
rows=rows+1
end

$end
workbook.Save
workbook.Close

$end
sleep 2

browser.close

File: Regression/NRG/Landing/Landing_Links_Logo.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.select import Select
import time
import xlrd
import os

if os.environ.get('USE_PHANTOM'):
    print("using PhantomJS")

    driver = webdriver.PhantomJS()
    driver.implicitly_wait(5)
    driver.set_window_size(1120, 550)
else:
    print("using Firefox")

driver = webdriver.Firefox()   

workbook = xlrd.open_workbook("./nrg_Links.xlsx")
worksheet = workbook.sheet_by_name('Landing')

for current_row in range(1,worksheet.nrows):
    page = worksheet.row(current_row)[0].value
    xpath = worksheet.row(current_row)[1].value
    expected_url = worksheet.row(current_row)[2].value
    link = worksheet.row(current_row)[4].value
 
    driver.get(page)
    time.sleep(2)

    elem = driver.find_element_by_xpath(xpath).click()
    try:
        driver.switch_to.alert.accept()
    except:
        pass
 
    if driver.current_url == expected_url:
        print ("Passed - ", link)
    else:
        print ("Failed - ", link)

driver.close()